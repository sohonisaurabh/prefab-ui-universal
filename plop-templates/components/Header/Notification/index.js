/**@jsx jsx */
import React from "react";
import { observer } from "mobx-react";
import { jsx, css } from "@emotion/core";

const notifBox = css`
  margin: 2rem 0 0 0;
  padding-right: 2em;
  position: relative;
  width: 50px;
`;
const cartCountBox = css`
  position: absolute;
  right: 30px;
  top: -22px;
  text-align: center;
  width: 25px;
  height: 25px;
  border-radius: 50px;
  border: 1px solid;
  background: white;
`;
const cartCount = css`
  display: inline-block;
  padding: 4px 0;
`;

const Notification = observer(({ store }) => {
  return (
    <div css={notifBox}>
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAQAAABLCVATAAAA+ElEQVR4AeXPMSjEYRjH8SfKIqOUY5DMyg7KZGGwT0hsEmW6wmoyKtkX+3STvWQw2QzEIJJ0PsYb7rzv++9u4vOM3/r1vvGnqQO47NXQl9HeDHEW3bKkukfj0c6N6vajnTVVNU1HO33uVXMRnVlXxYdadGbAo3In8Tt7Sj0bTA0NeVVmJ9IcK3GnP9IM+5S3GnnO5FxHCbNyZqKEKynfdqOEOdB0pN7xFqOEfrfgPLpjBXwYie44AA/q2VuIFMtK1SPNaflQhnmHXX+th0xoeNMwla8Jap4AT2q5mmBDy3auJtjUspWrCca8tB6frwkmNbxrmMjXf+IHAnWgsliFmvYAAAAASUVORK5CYII="
        alt=""
      />
      <div css={cartCountBox}>
        <span css={cartCount}>{store.getLikedItemsCount}</span>
      </div>
    </div>
  );
});

export default Notification;
