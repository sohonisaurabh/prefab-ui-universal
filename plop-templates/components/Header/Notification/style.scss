.notif-box {
    padding-right: 2em;
    position: relative;
    width: 50px;
}
.cart-count-box {
    position: absolute;
    right: 30px;
    top: -22px;
    text-align: center;
    width: 25px;
    height: 25px;
    border-radius: 50px;
    border: 1px solid;
    background: white;
}
.cart-count {
    display: inline-block;
    padding: 4px 0;
}