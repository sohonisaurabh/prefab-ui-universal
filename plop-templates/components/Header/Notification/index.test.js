import React from 'react';
import renderer from 'react-test-renderer';
import Notification from './index';

it('Notification renders correctly', () => {
  const store = {
    getLikedItemsCount: () => 8,
  };
  const tree = renderer.create(<Notification store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
});
