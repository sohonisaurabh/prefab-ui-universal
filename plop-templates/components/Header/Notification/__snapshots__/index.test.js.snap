// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<div
  className="css-1b4mhmc"
>
  <img
    alt=""
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAQAAABLCVATAAAA+ElEQVR4AeXPMSjEYRjH8SfKIqOUY5DMyg7KZGGwT0hsEmW6wmoyKtkX+3STvWQw2QzEIJJ0PsYb7rzv++9u4vOM3/r1vvGnqQO47NXQl9HeDHEW3bKkukfj0c6N6vajnTVVNU1HO33uVXMRnVlXxYdadGbAo3In8Tt7Sj0bTA0NeVVmJ9IcK3GnP9IM+5S3GnnO5FxHCbNyZqKEKynfdqOEOdB0pN7xFqOEfrfgPLpjBXwYie44AA/q2VuIFMtK1SPNaflQhnmHXX+th0xoeNMwla8Jap4AT2q5mmBDy3auJtjUspWrCca8tB6frwkmNbxrmMjXf+IHAnWgsliFmvYAAAAASUVORK5CYII="
  />
  <div
    className="css-q3u3xk"
  >
    <span
      className="css-p6in9o"
    />
  </div>
</div>
`;
